cirq/ops/eigen_gate.py

```diff
@@ -134,8 +134,8 @@ def matrix(self) -> np.ndarray:
         if self.is_parameterized():
             raise ValueError("Parameterized. Don't have a known matrix.")
         e = cast(float, self._exponent)
-        return np.sum(1j**(half_turns * e * 2) * component
-                      for half_turns, component in self._eigen_components())
+        return np.sum([1j**(half_turns * e * 2) * component for half_turns,
+                       component in self._eigen_components()], axis=0)
```

Comment:
Is the axis argument required?

Reply:
Yes, as otherwise np.sum will sum over all axes, which results in a scalar.